### ISSUE TITLE
Safari viewing "issue"

### ISSUE THREAD

**Issue by finch6** [thingwala/geyserwala-connect#1]
Hi,

I understand this is more of an Apple issue ― or iOS 18 Public Beta issue ― but just wondering if anyone else is experiencing this'd knows how to fix it?

Safari on my iPhone keeps giving me the following error:
```
Safari can't open the page.
The error was: "cannot parse response".
```
![IMG_9075A42B2E72-1](https://github.com/user-attachments/assets/da53487c-b665-4e83-92cf-972f52d4eeaa)

Anyone else having this issue?

It was working fine on iOS 17 Public Beta versions, so something has changed in iOS 18. I can still access `geyeserwala.local` on my Mac on Safari, so definitely is an iOS 18 issue.

Thanks in advance 👍🏼

**Comment by thingwala** [comment]
@finch6 Please provide more technical details of the issue. We are unable to reproduce using iPhone 15 (iOS 18.0 Beta 3) sim.

As a workaround you may want to try Google Chrome: https://apps.apple.com/us/app/google-chrome/id535886823

**Comment by finch6** [comment]
Sorry for the late reply.

This issue is still persistent on Safari ― I'm assuming it's due to the insecure site? I tried on Chrome and it works, but I get "Site not secure" messages in Chrome. Im assuming Safari is just being more secure than Chrome...

But either way, its not woking on Safari on Mac or iOS.
Mac ― 15.0.1 (24A348)
iOS ― 18.1

**Comment by thingwala** [comment]
The Local app is on HTTP, Web app is on HTTPS.
Issue report "its not woking on Safari on Mac or iOS" needs more detail.
Geyserwala firmware version?

**Comment by finch6** [comment]
Im not too sure what more detail you want???  The issue I mentioned in the beginning is persisting..

Im on the latest Geyserwala firmware as I have to install from the Geyserwala unit ― Version 20240915-ca71f3e

Im still getting the:
`Safari can't open the page.
The error was: "cannot parse response".`
when trying to navigate to `http://geyserwala.local` from my iPhone. This the iPhone 14 Pro that has iOS 18.1 on it.
Im getting there same behaviour on another iPhone 14 Pro that has iOS 18.1 on it.
Here is a screenshot of said behaviour:
![IMG_0621](https://github.com/user-attachments/assets/b1910c72-545d-4537-9d99-e6dd5aa39aa6)


When trying to navigate to `http://geyserwala.local` from my MacBook it just hangs.... This is the MacBook that has MacOS 15.0.1 (24A348) installed.
Here is a screenshot of the loading bar, it just gets stuck here:
<img width="834" alt="Screenshot 2024-10-08 at 6 51 25 PM" src="https://github.com/user-attachments/assets/9f82c75a-e58f-4506-8fd1-07cd69b1bcb8">


When trying to navigate to `192.168.0.20` from my MacBook it just hangs.... This is the MacBook that has MacOS 15.0.1 (24A348) installed.
Here is a screenshot of the loading bar, it just gets stuck here:
<img width="818" alt="Screenshot 2024-10-08 at 7 03 13 PM" src="https://github.com/user-attachments/assets/564eb422-d48e-4cba-a4a5-1de3eb863147">


The Geyserwala is definitely on my network:
<img width="911" alt="Screenshot 2024-10-08 at 7 00 37 PM" src="https://github.com/user-attachments/assets/9b1b14bb-7ff7-4187-a866-8e05999e58ab">


The only way I can access my Geyserwala is by using the Web app or by using Chrome:
<img width="1132" alt="Screenshot 2024-10-08 at 7 16 05 PM" src="https://github.com/user-attachments/assets/ba69f2d0-1ca2-4fea-80c4-0423c0203b3d">


Even when scanning the QR code the for the local app, it still gives me the `Safari can't open the page.
The error was: "cannot parse response".` error.

So I don't know, I'm lost for words 🤷🏼‍♂️

Im assuming this is a Apple issue???

**Comment by thingwala** [comment]
Thanks for clarification. Apple has recently introduced some new web criteria, WC3 reports one issue for our app?

Would you mind looking around in: Safari -> Develop -> Web Inspector, in both the Network and Console tabs. Any low lever details like this is appreciated.

**Comment by thingwala** [comment]
Found a reproducer with: 20240915-ca71f3e, iPhone 15 Pro (Simulator), iOS 18.1
Investigating ...

**Comment by thingwala** [comment]
The web server library pads the chunk length with spaces.
Awaiting a fix.

**Comment by finch6** [comment]
Seems to be good to go again, I can access the geyserwala on `geyserwala.local` again 👍🏼

Not too sure if it was the latest geyserwala firmware update or the latest iOS update? 🤷🏼‍♂️

![IMG_0781](https://github.com/user-attachments/assets/d82a0640-2dce-4c49-b3f6-1e1ae0e70f9d)

**Comment by thingwala** [comment]
There was a compatibility issue with our webserver. Fixed in 20241020-ed2a50d release.
